### PR TITLE
Make trait fields readonly

### PR DESCRIPTION
**Pull request by pchote**

### CODE REVIEW

**Comment by Phrohdoh** [comment]
:+1:

File: OpenRA.Mods.Common/Effects/Contrail.cs [hunk 1]
```diff
@@ -36,9 +36,9 @@ class ContrailInfo : ITraitInfo, Requires<IBodyOrientationInfo>
 
 	class Contrail : ITick, IRender
 	{
-		ContrailInfo info;
-		ContrailRenderable trail;
-		IBodyOrientation body;
+		readonly ContrailInfo info;
+		readonly ContrailRenderable trail;
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Effects/Contrail.cs]: This seems to break contrails (example: badgers on ra shellmap).

**Comment by reaperrr** [comment]
:+1: after contrails are fixed.

File: OpenRA.Mods.Common/Effects/Contrail.cs (hunk 1 above)
Comment (pchote) [inline OpenRA.Mods.Common/Effects/Contrail.cs]: Oh yes - readonly mutable structs are bad news!

**Comment by pchote** [comment]
Fixed.

**Pull request merged**